unlock-app/src/components/interface/checkout/main/Confirm.tsx

```diff
@@ -31,17 +31,23 @@ export function Confirm({ checkoutService, communication }: Props) {
     // If not pessimistic, we can emit the transaction info right away
     // and pass the signed message as well
     if (!paywallConfig.pessimistic && hash) {
-      communication?.emitTransactionInfo({
-        hash,
-        lock,
-        metadata,
-        network,
-      })
-      communication?.emitUserInfo({
-        address: account,
-        signedMessage: messageToSign?.signature,
-      })
-      communication?.emitMetadata(metadata)
+      communication?.emitTransactionInfo(
+        {
+          hash,
+          lock,
+          metadata,
+          network,
+        },
+        paywallConfig
+      )
+      communication?.emitUserInfo(
+        {
+          address: account,
+          signedMessage: messageToSign?.signature,
+        },
+        paywallConfig
```

Comment:
The paywallconfig object is already passed when initializing so I am not sure it makes sense to change the signature